### DISCUSSION TITLE
DatePicker - allow selection outside month, expose "first day" and "last day" of calendar in inline mode

### DISCUSSION THREAD

**Discussion by darylteo** [primefaces/community#2456] [PrimeVue]
Use case - I calculate a list of available dates that should be selectable on the date picker, but need the first and last day of the calendar when the month is selected in inline view.

I can probably compute this, but would be handy if @month-change would simply reveal expose this information when the month is changed.

Edit: ok i just realised you can't select outside of the month either. This might be a nice to have enhancement altogether in that case.

**Comment by ibrahimkelly** [comment]
> Use case - I calculate a list of available dates that should be selectable on the date picker, but need the first and last day of the calendar when the month is selected in inline view.
> 
> I can probably compute this, but would be handy if @month-change would simply reveal expose this information when the month is changed.
> 
> Edit: ok i just realised you can't select outside of the month either. This might be a nice to have enhancement altogether in that case.

Does date range from date <a href="https://primevue.org/datepicker/#range">PrimeVue DateRange</a> can help ?
Please take a look at 👆🏽

**Reply by darylteo** [reply]
![Screenshot 2024-08-07 at 9 54 12 pm](https://github.com/user-attachments/assets/0c681f44-97a3-4da2-8cc9-088b0eb900c1)

Sorry I meant regarding this. Nothing to do with selecting date range.
